src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/DownloadManager.java

```diff
@@ -194,6 +199,10 @@ public Path download(
       }
     }
 
+    if (disableDownload) {
+      throw new IOException(String.format("Failed to download repo %s: download is disabled.", repo));
```

Comment:
Please add test cases to verify at least the following cases (`starlark_repository_test` looks like a logical location, but maybe there is something better):

1. When the flag is set, downloads don't work
2. When the flag is set, repositories in distdir are still accessible
3. Maybe: when the flag is unset, downloads work (although that case is probably covered by existing tests?)

I'd love if this flag to also affected `git_repository` and the like, but I guess that's not possible unless we put repository download actions into a sandbox which is probably too big a change for now :(

Reply:
Test cases added.

`git_repository` uses `repository_ctx.execute` under the hood so it's impossible for this change to have affect on it.

Reply:
Yeah, I don't think there is a lot we can do about that :(